_posts/2019-11-25-hhvm-4.33.markdown

```diff
@@ -0,0 +1,41 @@
+---
+title: "HHVM 4.33"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.33 is released! This release marks the end of support for 4.27;
+4.28&ndash;4.32 remain supported, as does the 4.8 LTS release.
+
+# Highlights
+
+- Added `hh_client --concatenate-all paths/ morepaths/` to create a single-file
+  version of a Hack code base, respecting required inclusion order (class
+  hierarchy), namespace declarations, and potentially-conflicting `use`
+  statements. Current limitations:
+  - all code most be namespaced; we expect this limitation to be removed in a
+    future release
+  - there must not be more than one `<<__EntryPoint>>` function; no entrypoints
+    is also acceptable
+- For developers working on the implementation of Hack itself (in rust/ocaml):
+  - the test suite now passes in open source builds
+  - the code generation tools for updating the parser and rust/ocaml FFI are now
+    usable in opensource builds
+
+# Breaking Changes
+
+- FIXMEs for errors that were dropped (as an internal implementation detail) are
+  now also dropped; this can lead to some errors moving closer to the
+  problematic expression, and fixes some cases of errors being reported in a
+  caller/callee file instead of the correct file
+- An error is now raised if a reified function call is made without all required
+  generics; for example, `foo<vec>()` now raises an error, as `vec` requires 
+  a generic type
+- Type structures for type aliases are now keyed by opaqueness and the alias
```

Comment:
@oulgen could you check this please? Previous one might also be relevant to you, but I think it's Millie's diff